mne/io/edf/edf.py

```diff
@@ -590,10 +590,14 @@ def _read_edf_header(fname, exclude):
                                 for ch in channels])[sel]
         prefiltering = [fid.read(80).decode().strip(' \x00')
                         for ch in channels][:-1]
-        highpass = np.ravel([re.findall(r'HP:\s+(\w+)', filt)
-                             for filt in prefiltering])
-        lowpass = np.ravel([re.findall(r'LP:\s+(\w+)', filt)
-                            for filt in prefiltering])
+        highpass = np.array(
+            [v for hp in [re.findall(r'HP:\s*([0-9]+[.]*[0-9]*)', filt)
+                          for filt in prefiltering] for v in hp]
```

Comment:
was using something different from np.ravel necessary to make it work? can we make the REGEXP constant variables at the top of the file to avoid duplications. Having a private function to do this filtering would also allow you to have a unit test.

Reply:
np.ravel would make an array of `list` objects when one of the elements was an empty list (e.g. if a channel had no filter information). This was originally why I looked into this, as some relatively new data I was using off of physionet had a mix of filtered and unfiltered channels.